gnosis/src/main/scala/edu/uci/ics/cloudberry/gnosis/IGeoIndex.scala

```diff
@@ -41,4 +41,32 @@ class USGeoJSONIndex() extends IGeoIndex {
   override def search(envelope: Envelope): Seq[IUSGeoJSONEntity] = {
     index.query(envelope).asScala.map(item => entities(item.asInstanceOf[Int]))
   }
+}
+
+class NYGeoJSONIndex() extends IGeoIndex {
```

Comment:
I think you may also need to change the `implicit builder: (Map[String, AnyRef], Geometry) => INYGeoJSONEntity` to `implicit builder: (Map[String, AnyRef], Geometry) => IEntity`